materials/spatial-data-R.md

```diff
@@ -0,0 +1,132 @@
+---
+layout: page
+element: notes
+title: Spatial Data Introduction
+language: R
+--- 
+
+> Remember to download and set-up directory:
+>
+> * [LiDAR rasters and plots]({{ site.baseurl }}/data/Neon-airborne.zip)
+> * [`HARV_NDVI`]({{ site.baseurl }}/data/HARV_NDVI.zip) 
+> * [`SJER_NDVI`]({{ site.baseurl }}/data/SJER_NDVI.zip)
+
+> Set-up R Console:
+
+```
+library(raster)
+library(rgdal)
+```
+
+### `raster` structure, import, and plotting
+
+* Spatial data is often organized in a `raster`.
+    * gridded format (*like coordinates*)
+    * each pixel is a value
+* Metadata is important to describe the context of spatial data.
+    * `bands`
+    * `projection`
+    * `units`
+    * `min`, `max`, `mean`
+
+```
+> GDALinfo("HARV_dsmCrop.tif")
+rows        1367 
+columns     1697 
+bands       1 
+lower left origin.x        731453 
+lower left origin.y        4712471 
+res.x       1 
+res.y       1 
+ysign       -1 
+oblique.x   0 
+oblique.y   0 
+driver      GTiff 
+projection  +proj=utm +zone=18 +datum=WGS84 +units=m +no_defs 
+file        HARV_dsmCrop.tif 
+apparent band summary:
+   GDType hasNoDataValue NoDataValue blockSize1 blockSize2
+1 Float64           TRUE       -9999          1       1697
+apparent band statistics:
+    Bmin   Bmax    Bmean      Bsd
+1 305.07 416.07 359.8531 17.83169
+Metadata:
+AREA_OR_POINT=Area 
+```
+
+* `raster()` import
+
+```
+> dsm_harv <- raster("HARV_dsmCrop.tif")
+> dsm_harv
+class       : RasterLayer 
+dimensions  : 1367, 1697, 2319799  (nrow, ncol, ncell)
+resolution  : 1, 1  (x, y)
+extent      : 731453, 733150, 4712471, 4713838  (xmin, xmax, ymin, ymax)
+coord. ref. : +proj=utm +zone=18 +datum=WGS84 +units=m +no_defs +ellps=WGS84 +towgs84=0,0,0 
+data source : /Users/Zack/Desktop/DataRA/misc-datacarp/Neon-phenology-from-space/NEON-airborne/HARV_dsmCrop.tif 
+names       : HARV_dsmCrop 
+values      : 305.07, 416.07  (min, max)
+```
+
+* `plot(raster)`
+    * package modifies R basic `graphics`
+
+```
+plot(dsm_harv, main="HARV", xlab="Easting", ylab="Northing")
+```
+
+### `raster` math
+
+```
+dtm_harv <- raster("HARV_dtmCrop.tif")
+chm_harv <- dsm_harv - dtm_harv
+```
+
+> Do [Exercise 1 - Phenology from Space]({{ site.baseurl }}/exercises/Neon-phenology-from-space-R), Tasks 1-2.
+
+### Import and reproject shapefiles
+
+* `vector` data can be added to `rasters` to provide reference information
+
+```
+plots_harv <- readOGR("plot_locations/", "HARV_plots")
```

Comment:
Might be good to explain how shapefiles are special in that they consist of multiple files that have the same basename but different extensions, but the command to read them points only to the basename